### DISCUSSION TITLE
playing video file from local filesystem

### DISCUSSION THREAD

**Discussion by remi-grumeau** [neutralinojs/neutralinojs#682] [Q&A]
Hello,
Discovering Neutralino today, coming from NwJS.
Is there a reason why a static mp4 or webm file won't play?
The two files (mp4 & webm) are right next to the index.html.

I've tried to load it remotely from another computer (http://192.x.x ...) and it doesn't work neither.

Console says :
`Unhandled Promise Rejection: NotSupportedError: The operation is not supported`

Any help please?

**Comment by remi-grumeau** [comment]
Some additionnal informations
I'm testing on Ubuntu 20, using 'neu run'.
I get this output in the terminal.

```
Starting process: neutralino-linux_x64  --load-dir-res --path=. --debug-mode
devAPIServer listening at http://localhost:5050
ERROR 2021-11-01 00:35:09,708 Unable to open ./resources/briefing.mp4 api/debug/debug.cpp:15 root@unknown-host

ERROR 2021-11-01 01:12:09,288 Unable to open ./resources/briefing.webm api/debug/debug.cpp:15 root@unknown-host
```

I used to do the same thing (playing a video from a http source) directly in brave using --kiosk & --allow-running-insecure-content flags.

**Comment by shalithasuranga** [comment]
Hey.. I've just tried the following code on the console, and it worked fine.
```
window.location.href = `http://localhost:${NL_PORT}/resources/videos/test.mp4`
```
And this as well in `index.html`

```html
    <video width="320" height="240" controls>
        <source src="./videos/test.mp4" type="video/mp4">
    </video>
```

But, setting the URL from config didn't work. Maybe a bug in webview? Not sure. Will you be able to try above solutions? I think the `video` tag is a better approach.

**Reply by remi-grumeau** [reply]
Nope, still nothing.
Video does not load, HTML5 controls shows nothing load. Nothing neither in the inspector network.
No more error in the terminal neither when ask it to play.

But if i change mp4 to webm, then it shows in the inspector network, but then shows the error again.
![Capture d’écran 2021-11-01 à 15 24 34](https://user-images.githubusercontent.com/813438/139687652-6ea59f41-e242-42cc-8d77-c2c64b7e7e12.png)
![Capture d’écran 2021-11-01 à 15 25 26](https://user-images.githubusercontent.com/813438/139687659-2a7eaaa8-3920-4b07-8970-0c20aac86dec.png)

In case it helps, here is the ZIP of the full project
https://remi-grumeau.com/files/public/c38c9fed3701096504ba880570063b9c

**Reply by shalithasuranga** [reply]
Looks like you need to use `./briefing.mp4"` as the file name because you are already on the `www` base path.

**Reply by remi-grumeau** [reply]
Nope, still not doing it.
I did try
- ./briefing.mp4
- ../briefing.mp4
- /www/briefing.mp4
- ./www/briefing.mp4
- ../www/briefing.mp4

Since i have an apache server running on this machine too, i putted briefing.mp4 at the document root of this server, and set http://localhost/briefing.mp4 & http://its_ip/briefing.mp4
That's still a nogo.
(yet this video plays well in Brave at this same URL)

![Capture d’écran 2021-11-01 à 16 45 10](https://user-images.githubusercontent.com/813438/139699632-7170ddaa-a79e-42e7-bfde-e0b7a34993ac.png)

**Reply by shalithasuranga** [reply]
Looks like an issue with webview if it works with this command on your web browser
```
neu run --mode=browser
```

**Reply by remi-grumeau** [reply]
`neu run --mode=browser`
Works fine.
Using "./briefing.mp4" or just "briefing.mp4" as src="" for `<source>`, or even by just setting the "src" attribute to the video tag.
`<video width="100%" height="100%" controls autoplay src="briefing.mp4"></video>`

![Capture d’écran 2021-11-01 à 22 35 20](https://user-images.githubusercontent.com/813438/139745320-b5ac7f90-6393-4291-83a6-891dd7bf3007.png)

**Comment by Blatman** [comment]
@shalithasuranga  - FWIW I have found issues similar with the latest 3.0.0 source.
Using 10.13.6 High Sierra Mac   Desktop app using window mode and video/audio HTML tags. Media files co-located with index.html.

Pre 3.0.0 -  MP4 OK, WebM FAIL, MP3 OK, Wav OK

Latest 3.0.0 - MP4 FAIL, WebM FAIL, MP3 OK, Wav FAIL

Pre 3.0.0
<img width="646" alt="300pre" src="https://user-images.githubusercontent.com/11089335/139737618-2e55b4df-d7ea-42ed-966b-2af628567436.png">

Latest 3.0.0
<img width="650" alt="300latest" src="https://user-images.githubusercontent.com/11089335/139737729-733cdf66-1a43-495a-98e6-fc4f48eb7e64.png">
